File: backend/src/main/java/com/backend/domain/analysis/repository/AnalysisResultRepository.java
package com.backend.domain.analysis.repository;

import com.backend.domain.analysis.entity.AnalysisResult;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface AnalysisResultRepository extends JpaRepository <AnalysisResult, Long>{
    List<AnalysisResult> findAnalysisResultByRepositoriesIdOrderByCreateDateDesc(Long repositoryId);
    long countByRepositoriesId(Long repositoryId);
    List<AnalysisResult> findByRepositoriesId(Long repositoriedId);
    Optional<AnalysisResult> findTopByRepositoriesIdOrderByCreateDateDesc(Long repositoryId);
}

File: backend/src/main/java/com/backend/domain/analysis/repository/AnalysisResultRepository.kt
package com.backend.domain.analysis.repository

import com.backend.domain.analysis.entity.AnalysisResult
import com.backend.domain.repository.entity.Repositories
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface AnalysisResultRepository : JpaRepository<AnalysisResult, Long> {
    fun findAnalysisResultByRepositoriesIdOrderByCreateDateDesc(repositoryId: Long): List<AnalysisResult>
    fun countByRepositoriesId(repositoryId: Long): Long
    fun findByRepositoriesId(repositoryId: Long): List<AnalysisResult>
    fun findTopByRepositoriesIdOrderByCreateDateDesc(repositoryId: Long): AnalysisResult?
    fun findTopByRepositoriesOrderByCreateDateDesc(repositories: Repositories): AnalysisResult?
}
